### PR TITLE
Basic java.lang.Thread support

### PR DESCRIPTION
**Pull request by theresa-m**
This is intended to implement just enough to be able to initialize, start, and join a Java thead.

### CODE REVIEW

**Review by DanHeidinga — COMMENTED** [review]
Initial comments - still looking at parts of this

File: plugins/intrinsics/src/main/java/org/qbicc/plugin/intrinsics/core/CoreIntrinsics.java [hunk 1]
```diff
@@ -285,15 +289,92 @@ public static void registerJavaLangThreadIntrinsics(CompilationContext ctxt) {
...
+            /* set current thread */
+            ValueHandle qbiccCurrentThreadPointer = builder.pointerHandle(ctxt.getCurrentThreadLocalSymbolLiteral());
+            builder.store(qbiccCurrentThreadPointer, thread, MemoryAtomicityMode.NONE);
+
+            /* call "run" method of thread object */
+            VirtualMethodElementHandle runHandle = (VirtualMethodElementHandle)builder.virtualMethodOf(thread, jltDesc, "run", voidDesc);
+            builder.call(runHandle, List.of());
+
+            /* set java.lang.Thread.threadStatus to terminated */
+            Value terminatedState = builder.load(builder.staticField(jimVMDesc, "JVMTI_THREAD_STATE_TERMINATED", BaseTypeDescriptor.I), MemoryAtomicityMode.NONE);
```
Comment (DanHeidinga) [inline plugins/intrinsics/src/main/java/org/qbicc/plugin/intrinsics/core/CoreIntrinsics.java:328]: I think we can also hardcode `JVMTI_THREAD_STATE_TERMINATED` as the value is from the jvmti spec

File: plugins/intrinsics/src/main/java/org/qbicc/plugin/intrinsics/core/CoreIntrinsics.java [hunk 2]
```diff
@@ -285,15 +289,92 @@ public static void registerJavaLangThreadIntrinsics(CompilationContext ctxt) {
...
+        intrinsics.registerIntrinsic(vmHelpersDesc, "threadWrapper", threadWrapperDesc, threadWrapper);
+
+        /* private native void start0(); */
+        MethodDescriptor JLT_start0Desc = MethodDescriptor.synthesize(classContext, pthreadPtrDesc, List.of(voidPtrDesc));
+        InstanceIntrinsic start0 = (builder, instance, target, arguments) -> {
+            ValueHandle threadReference = builder.referenceHandle(instance);
+
+            /* set java.lang.Thread.threadStatus to runnable and alive */
+            Value runnableState = builder.load(builder.staticField(jimVMDesc, "JVMTI_THREAD_STATE_RUNNABLE", BaseTypeDescriptor.I), MemoryAtomicityMode.NONE);
+            Value aliveState = builder.load(builder.staticField(jimVMDesc, "JVMTI_THREAD_STATE_ALIVE", BaseTypeDescriptor.I), MemoryAtomicityMode.NONE);
```
Comment (DanHeidinga) [inline plugins/intrinsics/src/main/java/org/qbicc/plugin/intrinsics/core/CoreIntrinsics.java:343]: Another hard code candidate

**Comment by DanHeidinga** [comment]
We had an offline conversation about this PR - ping me when you're ready for another review pass

**Comment by theresa-m** [comment]
I will submit this for review in smaller pieces.

**Pull request closed without merging**